DOCSUP-3922: Document the untuple() function (#17110)
ClickHouse/ClickHouse: docs/en/sql-reference/functions/untuple-functions.md

---
toc_priority: 66

> toc_title: Untuple

Reviewer: ```suggestion
toc_title: Tuples
```

---

> # Untuple Functions {#untuple-functions}

Reviewer: ```suggestion
# Functions for Working with Tuples {#tuple-functions}
```

## Untuple {#untuple}

> Performs syntactic substitution of [tuple](../../sql-reference/data-types/tuple.md#tuplet1-t2) elements in the call location.

Reviewer: ```suggestion
Substitutes of [tuple](../../sql-reference/data-types/tuple.md#tuplet1-t2) elements in the call location.
```

**Syntax**

``` sql
untuple(x)
```

You can use the `EXCEPT` expression to skip columns as a result of the query.

**Parameters**

-   `x` - A `tuple` function, column, or tuple of elements. [Tuple](../../sql-reference/data-types/tuple.md).

**Returned value**

> -   None.

Reviewer: ```suggestion
-   Tuple elements as individual values.
```

**Examples**

Input table:

``` text
┌─key─┬─v1─┬─v2─┬─v3─┬─v4─┬─v5─┬─v6────────┐
│   1 │ 10 │ 20 │ 40 │ 30 │ 15 │ (33,'ab') │
│   2 │ 25 │ 65 │ 70 │ 40 │  6 │ (44,'cd') │
│   3 │ 57 │ 30 │ 20 │ 10 │  5 │ (55,'ef') │
│   4 │ 55 │ 12 │  7 │ 80 │ 90 │ (66,'gh') │
│   5 │ 30 │ 50 │ 70 │ 25 │ 55 │ (77,'kl') │
└─────┴────┴────┴────┴────┴────┴───────────┘
```

> Example of using a `tuple` function as a parameter `untuple` function:

Reviewer: This is the very strange example that looks like a test case. I think we can delete it. It doesn't give a user valuable information.

Query:

``` sql
SELECT untuple(tuple(v2,v3)) FROM kv;
```

Result:

``` text
┌─v2─┬─v3─┐
│ 20 │ 40 │
│ 65 │ 70 │
│ 30 │ 20 │
│ 12 │  7 │
│ 50 │ 70 │
└────┴────┘
```

> Example of using a column `Tuple` type as a parameter `untuple` function:

Reviewer: ```suggestion
Example of using a `Tuple`-type column as the `untuple` function parameter:
```

Query:

``` sql
SELECT untuple(v6) FROM kv;
```

Result:

``` text
┌─_ut_1─┬─_ut_2─┐
│    33 │ ab    │
│    44 │ cd    │
│    55 │ ef    │
│    66 │ gh    │
│    77 │ kl    │
└───────┴───────┘
```

Example of using an `EXCEPT` expression:

Query:

``` sql
SELECT untuple((* EXCEPT (v2, v3),)) FROM kv;
```

> Result:

Reviewer: I haven't understand the result. Why `v6` wasn't untupled?
